cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HTableNameConverter.java

```diff
@@ -16,18 +16,113 @@
 
 package co.cask.cdap.data2.util.hbase;
 
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.data2.util.TableId;
+import co.cask.cdap.proto.Id;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
 /**
  * Common utility methods for dealing with HBase table name conversions.
  */
 public abstract class HTableNameConverter {
```

Comment:
Feels like many methods of this class should be package private? Ideally they are not used outside of the HBaseTableUtil and CoProcessor (some method may need to be public because of that, but we should document it clearly that methods of this class shouldn't be used outside).